superdesk/io/feeding_services/twitter.py

```diff
@@ -0,0 +1,164 @@
+    label = 'Twitter'
+
+    ERRORS = [IngestTwitterError.TwitterLoginError().get_error_description(),
+              IngestTwitterError.TwitterNoScreenNamesError().get_error_description()]
+
+    def _test(self, provider):
+        self._update(provider, update=None, test=True)
+
+    def _update(self, provider, update, test=False):
+        config = provider.get('config', {})
+        consumer_key = config.get('consumer_key', '')
+        consumer_secret = config.get('consumer_secret', '')
+        access_token_key = config.get('access_token_key', '')
+        access_token_secret = config.get('access_token_secret', '')
+        screen_names = config.get('screen_names', '')
+        status_count = 100
+        # how many statuses to get, 200 should be max
+
+        new_items = []
+        api = twitter.Api(consumer_key=consumer_key,
+                          consumer_secret=consumer_secret,
+                          access_token_key=access_token_key,
+                          access_token_secret=access_token_secret)
+        try:
+            screen_names = screen_names.split(',')
+        except Exception as ex:
+            raise IngestTwitterError.TwitterNoScreenNamesError(ex, provider)
+
+        if not screen_names:
+            raise IngestTwitterError.TwitterNoScreenNamesError(ex, provider)
+        for screen_name in screen_names:
+            screen_name = screen_name.replace(' ', '')
+            try:
+                # hashtag search
+                if screen_name.startswith('#'):
+                    statuses = api.GetSearch(screen_name.lstrip('#'), count=status_count)
+                # user search
+                else:
+                    statuses = api.GetUserTimeline(screen_name=screen_name, count=status_count)
+            except twitter.error.TwitterError as exc:
+                if exc.message[0].get('code') == 34:
+                    # that page does not exist
+                    continue
+                elif exc.message[0].get('code') == 32:
+                    raise IngestTwitterError.TwitterLoginError(exc, provider)
+
+            for status in statuses:
+                d = parser.parse(status.created_at)
+                guid_hash = hashlib.sha1(status.text.encode('utf8')).hexdigest()
+                guid = generate_guid(type=GUID_TAG, id=guid_hash)
+                headline = "%s: %s" % (status.user.screen_name, status.text)
+                item = {}
+                item['headline'] = headline
+                item['type'] = 'text'
+                item['guid'] = guid
+                item['versioncreated'] = d
+                item['firstcreated'] = d
+
+                item['body_html'] = status.text
+                # include URL on body
+                urls = re.findall('http[s]?://(?:[a-zA-Z]|[0-9]|[$-_@.&+]|[!*\(\),]|(?:%[0-9a-fA-F][0-9a-fA-F]))+', status.text)
+                if urls:
+                    item['body_html'] += '<p><a href="%s" target="_blank">%s</a></p>' % (urls[0], urls[0])
```

Comment:
I know that dangerous chars are not in the current regex, but I would be more comfortable with some quoting, just in case somebody touch the regex in the future, to avoid XSS.